src/rebuild/srv.c

```diff
@@ -2117,10 +2069,17 @@ rebuild_tgt_prepare(crt_rpc_t *rpc, struct rebuild_tgt_pool_tracker **p_rpt)
 
 	pool = ds_pool_lookup(rsi->rsi_pool_uuid);
 	if (pool == NULL) {
-		D_ERROR("Can not find pool.\n");
+		D_ERROR("Can not find pool. "DF_UUID"\n", DP_UUID(rsi->rsi_pool_uuid));
 		return -DER_NONEXIST;
 	}
 
+	if (ds_pool_get_version(pool) < rsi->rsi_rebuild_ver) {
+		D_INFO(DF_UUID" map %u < rsi_rebuild_ver %u\n",
+		       DP_UUID(rsi->rsi_pool_uuid), ds_pool_get_version(pool),
+		       rsi->rsi_rebuild_ver);
+		D_GOTO(out, rc = -DER_BUSY);
```

Comment:
not a problem of this patch, just confirm that in this fail case it will cause on rebuild leader node -in rebuild_task_ult(), rebuild_leader_start() fail, it will do below for err handling -
1301 >       >       rc = rebuild_iv_update(pool->sp_iv_ns, &iv, CRT_IV_SHORTCUT_NONE,                   
1302 >       >       >       >              CRT_IV_SYNC_LAZY, true);  
then reschedule it by -
1329 >       >       ret = ds_rebuild_schedule(pool, task->dst_map_ver,                                  
1330 >       >       >       >       >         &task->dst_tgts, opc, 5);

But it is possible that the async lazy rebuild_iv_update possibly reach other server even after the re-scheduled rebuild get executed? (although very rare).
maybe can change to EAGER mode for the rebuild_iv_update()?

Reply:
hmm, probably we should update the rebuild iv synchronous for failure case, I remember there are some issues to block here, but forgot it. :(